packages/loader/container-loader/src/container.ts

```diff
@@ -811,6 +811,8 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                     this.context.notifyAttaching();
                 }
 
+                const abortController = new AbortController();
+                abortController.signal.addEventListener("abort", (event: Event) => { this.close(); });
```

Comment:
What this code does is closes container if someone cancelled this signal. But nobody will cancel it, so this code does nothing.
Instead (based on your description) what you suggest to do is to listen on "closed" event on container and in response to it cancel this abort controller.